### ISSUE TITLE
Recursive Code Doesn't seem to work in Python?

### ISSUE THREAD

**Issue by elvinup** [twibiral/obsidian-execute-code#154]
![image](https://user-images.githubusercontent.com/14940552/197446398-58c640bb-128a-422d-ab17-88bf3f2a6833.png)

Is it just me, or is it the recursion in this function that's causing it to not recognize an already defined function?

**Comment by chlohal** [comment]
I'm not sure, but this seems to be the same root problem as #146, which is fixed in #147. 

If you would like to apply the patch for yourself before the PR gets merged & published, you can do so as follows:

On line 12054 and 12059 of `<vault folder>/.obsidian/plugins/execute-code/main.js`, remove "`, ${localsName}`"

```diff
          ${printName}(eval(
              compile(${JSON.stringify(code.replace(/\r\n/g, "\n") + "\n")}, "<code block>", "eval"),
-             ${globalsName}, ${localsName}
+             ${globalsName}
          ))
      except SyntaxError:
          exec(
              compile(${JSON.stringify(code.replace(/\r\n/g, "\n") + "\n")}, "<code block>", "exec"),
-             ${globalsName}, ${localsName}
+             ${globalsName}
          )
```

**Comment by twibiral** [comment]
Fix released in [1.3.0](https://github.com/twibiral/obsidian-execute-code/releases/tag/1.3.0)
